api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -485,4 +489,23 @@ public void syncEraCommonsStatusUsingImpersonation(User user) throws IOException
       }
     }
   }
+
+  public void syncTwoFactorAuthStatus() {
+    syncTwoFactorAuthStatus(userProvider.get());
+  }
+
+  public void syncTwoFactorAuthStatus(User targetUser) {
+    updateUserWithRetries(user -> {
+      Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+      if (directoryService.getUser(user.getEmail()).getIsEnrolledIn2Sv()) {
+        user.setTwoFactorAuthCompletionTime(now);
```

Comment:
I believe this will overwrite the twoFactorAuthCompletionTime if it called on a user that already has it set. Is that what we want? This is different from what is happening in ProfileController. ProfileController only sets it if the value is originally null.

Side note: this is another reason for sharing the code here and in ProfileController. Less likelihood of differing implementation and bugs.

Reply:
good catch. will do.